app.rb

```diff
@@ -0,0 +1,175 @@
+require_relative 'student'
+require_relative 'teacher'
+require_relative 'rental'
+require_relative 'book'
+
+class App
+  def initialize
+    @people = []
+    @books = []
+    @id = 50
+  end
+
+  def run
+    selection = nil
+    puts 'Welcome to School Library App!'
+    while selection != 7
+      display_options
+      selection = gets.chomp.strip.to_i
+      puts
+      do_task(selection)
+    end
+    puts 'Thank you for using this app!'
+  end
+
+  def display_options
+    puts
+    puts 'Please choose an option by entering a number:'
+    puts '1 - List all books'
+    puts '2 - List all people'
+    puts '3 - Create a person'
+    puts '4 - Create a book'
+    puts '5 - Create a Rental'
+    puts '6 - List all rentals for a given person id'
+    puts '7 - Exit'
+  end
+
+  def do_task(selection)
+    case selection
+    when 1, 2, 6
+      list_tasks(selection)
+    when 3, 4, 5
+      create_tasks(selection)
+    when 7
+      nil
+    else
+      puts 'Please select a valid choice from the list'
+    end
+  end
+
+  def list_tasks(selection)
+    case selection
+    when 1
+      list_all_books
+    when 2
+      list_all_people
+    when 6
+      list_person_rentals
+    end
+  end
+
+  def create_tasks(selection)
+    case selection
+    when 3
+      create_person
+    when 4
+      create_book
+    when 5
+      create_rental
+    end
+  end
+
+  def create_person
+    print 'Do you want to create a student (1) or a teacher (2)? [Input the number]:'
+    person_selected = gets.chomp.strip.to_i
+    case person_selected
+    when 1
+      create_student
+      @id += 1
+    when 2
+      create_teacher
+      @id += 1
+    else
+      puts 'Invalid Selection. Returning to main menu'
+    end
+  end
+
+  def create_student
+    print 'Age: '
+    age = gets.chomp.to_i
+    if age <= 0
+      @id -= 1
+      return puts 'Wrong input for age. Returning to main menu'
+    end
+    print 'Name: '
+    name = gets.chomp.strip.capitalize
+    print 'Has Parent Permission? [Y/N]'
+    permission = gets.chomp.strip.upcase
+    case permission
+    when 'Y', 'YES'
+      permission = true
+    when 'N', 'NO'
+      permission = false
+    else
+      puts 'Wrong input for permission'
+      @id -= 1
+      return
+    end
+    student = Student.new(@id, age, name, parent_permission: permission)
+    @people << student
+    puts 'Person created successfully'
+  end
+
+  def create_teacher
+    print 'Age: '
+    age = gets.chomp.to_i
+    if age <= 0
+      @id -= 1
+      return puts 'Wrong input for age. Returning to main menu'
+    end
+    print 'Name: '
+    name = gets.chomp.strip.capitalize
+    print 'Specialization: '
+    specialization = gets.chomp.strip.capitalize
+    teacher = Teacher.new(@id, age, name, specialization)
+    @people << teacher
```

Comment:
- [ ] When I try to create a teacher, I get this error:

![image](https://user-images.githubusercontent.com/77106962/168174613-0e68c314-d377-42ce-8729-39a40b1939ce.png)

#### The error occurs because of the way you called the `super` method in your `Teacher` class. `Teacher` class inherits from the `Person` class and the initialize method for the `Person` class doesn't have a parameter for `id`. Kindly remove the `id` parameter of the `super` function in the `Teacher` class and you would be able to create a Teacher object successfully. 👍